framework/framework.go

```diff
@@ -150,6 +150,12 @@ func (f *Framework) DeleteFunc(obj interface{}) {
 		f.logger.LogCtx(ctx, "event", "delete", "function", "DeleteFunc", "level", "error", "message", "stop framework reconciliation due to error", "stack", fmt.Sprintf("%#v", err))
 		return
 	}
+
+	err = f.removeFinalizer(obj)
+	if err != nil {
+		f.logger.Log("event", "delete", "function", "DeleteFunc", "level", "error", "message", "stop framework reconciliation due to error", "stack", fmt.Sprintf("%#v", err))
+		return
+	}
```

Comment:
Did you check if removing finalizer generates object update event? If so this should be handled, possibly with additional annotation.

Reply:
I did not check, but I currently try to handle it by checking if `DeletionTimestamp` is set here:
https://github.com/giantswarm/operatorkit/blob/08fef601dfba839fe981b741de6770342470f946/framework/finalizer.go#L27

That way we would ignore any update events that happen on an object that is already marked for deletion and only handle delete events.

Reply:
Nice, let's hope it works :)